internal/bundle/helpers.go

```diff
@@ -68,3 +76,169 @@ func destroyBundle(t *testing.T, path string) error {
 	_, _, err := c.Run()
 	return err
 }
+
+var sparkVersions = []string{
+	"11.3.x-scala2.12",
+	"12.2.x-scala2.12",
+	"13.0.x-scala2.12",
+	"13.1.x-scala2.12",
+	"13.2.x-scala2.12",
+	"13.3.x-scala2.12",
+	"14.0.x-scala2.12",
+	"14.1.x-scala2.12",
+}
+
+func generateNotebookTasks(notebookPath string, nodeTypeId string) []jobs.SubmitTask {
+	tasks := make([]jobs.SubmitTask, 0)
+	for i := 0; i < len(sparkVersions); i++ {
+		task := jobs.SubmitTask{
+			TaskKey: fmt.Sprintf("notebook_%s", strings.ReplaceAll(sparkVersions[i], ".", "_")),
+			NotebookTask: &jobs.NotebookTask{
+				NotebookPath: notebookPath,
+			},
+			NewCluster: &compute.ClusterSpec{
+				SparkVersion:     sparkVersions[i],
+				NumWorkers:       1,
+				NodeTypeId:       nodeTypeId,
+				DataSecurityMode: compute.DataSecurityModeUserIsolation,
```

Comment:
this is only for Unity Catalog-enabled workspaces, but it doesn't check for .`DataSecurityModeSingleUser` and `DataSecurityModeNone`. 

what about non-uc workspaces, like `DataSecurityModeLegacySingleUser`?